webrender/src/batch.rs

```diff
@@ -1604,6 +1604,34 @@ impl Primitive {
             }
         }
     }
+
+    pub fn is_cacheable(
+        &self,
+        resource_cache: &mut ResourceCache
+    ) -> bool {
+        let image_key = match self.details {
+            PrimitiveDetails::Brush(BrushPrimitive { kind: BrushKind::Image{ request, ..  }, .. }) => {
+                Some(request.key)
+            }
+            PrimitiveDetails::Brush(BrushPrimitive { kind: BrushKind::YuvImage{ yuv_key, .. }, .. }) => {
+                Some(yuv_key[0])
+            }
+            PrimitiveDetails::Brush(_) |
+            PrimitiveDetails::TextRun(..) => {
+                None
+            }
+        };
+        if let Some(image_key) = image_key {
```

Comment:
One last nit - we could probably write this as something like:

```
        image_key.map_or(true, |image_key| {
            match resource_cache.get_image_properties(image_key) {
                Some(ImageProperties { external_image: Some(_), .. }) => {
                    false
                }
                Some(_) |
                None => {
                    true
                }
            }
        })
```

Which would remove the need for the last true and any early return statements. Then after squashing this should be good to merge, thanks!